### PR TITLE
Add OpenSearch to Rubin Cluster

### PR DESCRIPTION
**Pull request by Thysk**
This contains a basic values setup for master and worker nodes for an OpenSearch cluster to be deployed for monitoring purposes.

The makefile can be incorporated into the main makefile, and the values tweeked before the helm files being made and deployed

### CODE REVIEW

File: overlays/prod/opensearch/opensearch/kustomization.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+namespace: opensearch-system
+
+resources:
+- helm-opensearch-master.yaml
+- helm-opensearch-worker.yaml
+#- ns.yaml
```
Comment (s-sajid-ali) [inline overlays/prod/opensearch/opensearch/kustomization.yaml:6]: Could you elaborate on why `ns.yaml` is commented out? Thanks!

Comment (Thysk) [reply overlays/prod/opensearch/opensearch/kustomization.yaml:6]: It is commented out as I did not want the namespace effected by redeployment etc during development. And I was unsure if you wanted the namespace defied here or not

**Comment by s-sajid-ali** [comment]
Hitting the following error when deploying:
```
error: accumulating resources: accumulation err='accumulating resources from 'opensearch': '/Users/sasyed/Documents/packages/rubin-rucio-deploy/overlays/prod/opensearch' must resolve to a file': recursed accumulation of path '/Users/sasyed/Documents/packages/rubin-rucio-deploy/overlays/prod/opensearch': accumulating resources: accumulation err='accumulating resources from 'opensearch/': '/Users/sasyed/Documents/packages/rubin-rucio-deploy/overlays/prod/opensearch/opensearch' must resolve to a file': recursed accumulation of path '/Users/sasyed/Documents/packages/rubin-rucio-deploy/overlays/prod/opensearch/opensearch': accumulating resources: accumulation err='accumulating resources from 'helm-opensearch-master.yaml': evalsymlink failure on '/Users/sasyed/Documents/packages/rubin-rucio-deploy/overlays/prod/opensearch/opensearch/helm-opensearch-master.yaml' : lstat /Users/sasyed/Documents/packages/rubin-rucio-deploy/overlays/prod/opensearch/opensearch/helm-opensearch-master.yaml: no such file or directory': must build at directory: not a valid directory: evalsymlink failure on '/Users/sasyed/Documents/packages/rubin-rucio-deploy/overlays/prod/opensearch/opensearch/helm-opensearch-master.yaml' : lstat /Users/sasyed/Documents/packages/rubin-rucio-deploy/overlays/prod/opensearch/opensearch/helm-opensearch-master.yaml: no such file or directory
make: *** [run-apply] Error 1
```
Currently investigating the issue.

**Comment by s-sajid-ali** [comment]
Changed the location of the `values` file the `Makefile` for `opensearch` from `--values=opensearch/values-opensearch-worker.yaml` to `--values=opensearch/opensearch/values-opensearch-worker.yaml` to reflect their location and adding a slash in the `kustomization.yaml` for `opensearch` to reflect it's status as a directory.

With these changes, I now get the following error:
```
❯ make opensearch
helm repo add rucio https://rucio.github.io/helm-charts
"rucio" already exists with the same configuration, skipping
helm repo add elastic https://helm.elastic.co
"elastic" already exists with the same configuration, skipping
helm repo add opensearch https://opensearch-project.github.io/helm-charts/
"opensearch" already exists with the same configuration, skipping
helm repo update rucio elastic opensearch
Hang tight while we grab the latest from your chart repositories...
...Successfully got an update from the "opensearch" chart repository
...Successfully got an update from the "rucio" chart repository
...Successfully got an update from the "elastic" chart repository
Update Complete. ⎈Happy Helming!⎈
helm template opensearch opensearch/opensearch --version=2.13.0 --values=opensearch/opensearch/values-opensearch-master.yaml > opensearch/helm-opensearch-master.yaml
Error: Chart.yaml file is missing
make: *** [open-deployment] Error 1
```

Per [this StackOverflow answer](https://stackoverflow.com/a/69550583) (which refers to [this comment](https://github.com/hashicorp/terraform-provider-helm/issues/735#issuecomment-828779466), the error comes from upstream helm when the directory contains a directory with the same name as the chart name. 

Is there any particular reason to have the values files in `opensearch/opensearch` instead of just `opensearch`?

**Comment by Thysk** [comment]
You are right about the directory layout, and I will fix it, which in turn fixes the initial issue with wrong directory. It was a holdover from the elastic repo that I added too, but that repo was designed to have other things than just elasticsearch (e.g. logstash)

File: overlays/prod/Makefile [hunk 2]
```diff
@@ -25,6 +27,16 @@ elasticsearch:
...
 
+open-deployment:
+	helm template opensearch opensearch/opensearch --version=${OPEN_VERSION} --values=opensearch/values-opensearch-master.yaml > opensearch/helm-opensearch-master.yaml
+	helm template opensearch opensearch/opensearch --version=${OPEN_VERSION} --values=opensearch/values-opensearch-worker.yaml > opensearch/helm-opensearch-worker.yaml
+
+opensearch: helm open-deployment
+
+opensearch-apply:  opensearch run-apply
+
+opensearch-destry: opensearch run-destroy
```
Comment (s-sajid-ali) [inline overlays/prod/Makefile:38]: Typo, `destry` -> `destroy`

**Review by s-sajid-ali — COMMENTED** [review]
Could you also add the `opensearch` target to the `apply` target in the `overlays/prod/Makefile` (line 77)?

File: overlays/prod/Makefile (hunk 2 above)
Comment (Thysk) [reply overlays/prod/Makefile]: done
(I really need to check my o key, it is often not registering the presses recently)

**Comment by Thysk** [comment]
Added opensearch to apply target

**Comment by s-sajid-ali** [comment]
Thanks! 

The `usdf-hermes` pod is failing with:
```
  Warning  Failed     15s (x8 over 100s)  kubelet            Error: secret "usdf-elasticsearch-x-rucio-x-vcluster--usdf-rucio" not found
```

I believe this was fixed in https://github.com/slaclab/rubin-rucio-deploy/pull/22. Could you rebase this PR over that? The `main` branch might conflict with this PR due to https://github.com/slaclab/rubin-rucio-deploy/pull/26.

**Comment by Thysk** [comment]
not sure if removing the emptyDir from the securityPolicy will help or not, but its the only thing that I can find that references it

**Comment by Thysk** [comment]
Now that this is effectively deployed, is there a reason this isnt merged?

**Comment by bjwhite-fnal** [comment]
@Thysk Pretty much just waiting for you to say that.

**Pull request merged**